### DISCUSSION TITLE
absoluteDisplayStart vs absolutePlaybackStart

### DISCUSSION THREAD

**Discussion by mxAlexG** [CoderLine/alphaTab#961] [Q&A] [answered]
Hi,
How both properties differ from each other? I am trying to get the tick position value from a Beat instance. The problem however is that the value is useless when there are repeating sections in the score. I would appreciate a clarification.

Regards.

**Comment by Danielku15** [comment, marked as answer]
The difference is documented on the model: 
https://github.com/CoderLine/alphaTab/blob/4d8eac9140a2c4c53afdce5206c164f11a3b81e6/src/model/Beat.ts#L371-L381

It mainly relates to the special handling of grace notes within music notation. Their time when they are played is different from the time they are relatively displayed within the measure. 

When you generate a midi out of a score, a special lookup object `MidiTickLookup` is generated which allows a lookup of a bar/beat given a midi tick position. It can be obtained from the API object through the `api.tickCache`. As it is rather a low-level API I recommend looking through the code of alphaTab to understand all its details.

**Comment by mxAlexG** [comment]
Thanks for the clarification.
